test/scripts/e2e_subs/rest-participation-key.sh

```diff
@@ -9,33 +9,48 @@ date "+$0 start %Y%m%d_%H%M%S"
 # Use admin token for both get and post
 export USE_ADMIN=true
 
-pushd "${TEMPDIR}" || exit
+pushd "${TEMPDIR}" || exit 1
 
 FIRST_ROUND=0
 # A really large (but arbitrary) last valid round
-LAST_ROUND=1200000
+LAST_ROUND=120
 
 NAME_OF_TEMP_PARTKEY="tmp.${FIRST_ROUND}.${LAST_ROUND}.partkey"
 
 algokey part generate --first ${FIRST_ROUND} --last ${LAST_ROUND} --keyfile ${NAME_OF_TEMP_PARTKEY} --parent ${ACCOUNT}
 
-popd || exit
+popd || exit 1
 
 call_and_verify "Get List of Keys" "/v2/participation" 200 'address'
 
 # Find out how many keys there are installed so far
+if [[ $RES != *"address"*"effective-first-valid"* ]]; then
+    echo "Unexpected output: " $RES
+    exit 1
+fi
 NUM_IDS_1=$(echo "$RES" | python3 -c 'import json,sys;o=json.load(sys.stdin);print(len(o))')
+RES=""
 
 call_post_and_verify "Install a basic participation key" "/v2/participation" 200 ${NAME_OF_TEMP_PARTKEY} 'partId'
 
 # Get the returned participation id from the RESULT (aka $RES) variable
+if [[ $RES != *"partId"* ]]; then
+    echo "Unexpected output: " $RES
+    exit 1
+fi
 INSTALLED_ID=$(echo "$RES" | python3 -c 'import json,sys;o=json.load(sys.stdin);print(o["partId"])')
+RES=""
 
 # Should contain the installed id
 call_and_verify "Get List of Keys" "/v2/participation" 200 'address' "${INSTALLED_ID}"
 
 # Get list of keys
+if [[ $RES != *"address"*"effective-first-valid"* ]]; then
+    echo "Unexpected output: " $RES
+    exit 1
+fi
 NUM_IDS_2=$(echo "$RES" | python3 -c 'import json,sys;o=json.load(sys.stdin);print(len(o))')
+RES=""
 
 if [[ $((NUM_IDS_1 + 1)) -ne $NUM_IDS_2 ]]; then
```

Comment:
I think the way this test is using `NUM_IDS_*` is problematic. If someone added another test that installed a key, this one could become flaky.

We can remove all references to `NUM_IDS_*` and the assertions like this one that uses them. The other assertions check the for the IDs directly, so there's no need to have these catch-all checks.

i.e. after the key is installed, we query that the new key is there -- so no need to check that the number of keys is `NUM_IDS_1 + 1`.

Reply:
Can you please make a code change suggestion?